scrapy/linkextractors/regex.py

```diff
@@ -14,16 +14,25 @@ def clean_link(link_text):
     """Remove leading and trailing whitespace and punctuation"""
     return link_text.strip("\t\r\n '\"")
 
+
 class RegexLinkExtractor(SgmlLinkExtractor):
     """High performant link extractor"""
 
     def _extract_links(self, response_text, response_url, response_encoding, base_url=None):
+        def clean_text(text):
+            return replace_escape_chars(remove_tags(text.decode(response_encoding))).strip()
+
+        def clean_url(url):
```

Comment:
I just recalled thinking 'there is a reason I'm writing it this way' while rewriting this one.

What do you think about the line throwing anything else than `ValueError`?
Should it return `''` then, or the original url?